### PR TITLE
Add isRemote to the SpanContext

### PR DESCRIPTION
**Pull request by bogdandrutu**
Implements https://github.com/open-telemetry/opentelemetry-specification/pull/216

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/trace/SpanContext.java [hunk 1]
```diff
@@ -146,14 +175,21 @@ public String toString() {
         + spanId
         + ", traceFlags="
         + traceFlags
+        + ", isRemote="
+        + isRemote
```
Comment (jkwatson) [inline api/src/main/java/io/opentelemetry/trace/SpanContext.java]: just curious...why isn't this an autovalue class?

**Comment by jkwatson** [comment]
Is there a relevant spec or OTEP reference we can include with the PR?

**Comment by arminru** [comment]
@jkwatson 
> Is there a relevant spec or OTEP reference we can include with the PR?

I added a reference to https://github.com/open-telemetry/opentelemetry-specification/pull/216

File: api/src/main/java/io/opentelemetry/trace/SpanContext.java (hunk 1 above)
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/trace/SpanContext.java]: Good question :)) Probably it should be.

Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/trace/SpanContext.java]: https://github.com/open-telemetry/opentelemetry-java/issues/628

**Pull request merged**